sphinx/ext/intersphinx.py

```diff
@@ -112,18 +112,18 @@ def split_lines(iter):
             continue
         if location.endswith(u'$'):
             location = location[:-1] + name
-        location = join(uri, location)
+        location = posixpath.join(uri, location)
```

Comment:
This came from PR #3619, there's a comment there about not using `posixpath.join` to join filesystem paths on windows. However it appears to me that this join is only used for generating the paths that will be injected into the output, which I think should always be posix/URL-type paths regardless of the OS and if the files are local or remote (the browser or other program should convert them back I think).